### ISSUE TITLE
DEVICE_BUSY not reported to app

### ISSUE THREAD

**Issue by superafroman** [stripe/stripe-terminal-android#375]
## Summary
When operating in offline mode transaction processing is pretty quick. This has highlighted an issue for us when we call `collectPaymentMethod` too soon after another payment has been taken. The reader responds with a DEVICE_BUSY error, however this is only logged in the SDK, not reported to our app. We are left thinking everything is working but the reader remains inactive.

## Android version
Tested on Android 9 and 12

## Impacted devices (Android devices or readers)
WisePad 3

## SDK version
3.0.0

## Other information
Possibly related to https://github.com/stripe/stripe-terminal-android/issues/374 (error events occurring within SDK but not reported to app)

## Logs
```
2023-09-25 10:44:23.279  1980-1980  StripeTerminal          co.givealittle.kiosk                 I  class=BbposPaymentCollectionListener message="onReturnTransactionResult APPROVED"
2023-09-25 10:44:23.288  1980-1980  StripeTerminal          co.givealittle.kiosk                 I  class=PaymentCollectionCoordinator message="onHandlePaymentCollectionEvent TransactionCompletedEvent(result=APPROVED)"
2023-09-25 10:44:23.290  1980-2964  StripeTerminal          co.givealittle.kiosk                 I  class=StageEventLogger message="KernelProcessing newStatus IDLE event TransactionCompletedEvent"
2023-09-25 10:44:23.292  1980-2964  StripeTerminal          co.givealittle.kiosk                 I  class=PaymentCollectionCoordinator message="onHandlePaymentCollectionEvent PresentPaymentMethodEvent(model=CollectionModel(transactionType=CHARGE, amount=Amount(value=1000, currency=GBP), baseAmount=Amount(value=1000, currency=GBP), callToAction=null, readerActiveInterfaces=[ICC, NFC], readerContactCardSlotState=EMPTY, readerEnableEditTip=false, enableCustomerCancellation=false))"
2023-09-25 10:44:23.294  1980-2964  StripeTerminal          co.givealittle.kiosk                 D  class=BbposAdapterPaymentCollectionListener message=onRequestPaymentMethodCollection
2023-09-25 10:44:23.294  1980-2964  StripeTerminal          co.givealittle.kiosk                 I  class=TerminalStatusManager message="requestReaderInput Insert / Tap."
2023-09-25 10:44:23.295  1980-2964  StripeTerminal          co.givealittle.kiosk                 I  class=ProxyReaderListener message="onRequestReaderInput(Insert / Tap)"
2023-09-25 10:44:23.295  1980-2964  StripePaymentSession    co.givealittle.kiosk                 D  onRequestReaderInput: Insert / Tap
2023-09-25 10:44:23.296  1980-1980  StripeTerminal          co.givealittle.kiosk                 I  class=BbposControllerListener message="onError DEVICE_BUSY, Device busy with command/callback: onRequestOnlineProcess/onReturnBatchData/onReturnReversalData/onReturnTransactionResult"
2023-09-25 10:44:24.344  1980-1980  StripeTerminal          co.givealittle.kiosk                 I  class=BbposPaymentCollectionListener message="onReturnCheckCardResult NO_CARD cardData CardData{}"
2023-09-25 10:44:24.345  1980-1980  StripeTerminal          co.givealittle.kiosk                 I  class=PaymentCollectionCoordinator message="onHandlePaymentCollectionEvent ContactCardUpdateEvent(updatedCardState=EMPTY)"
2023-09-25 10:44:24.347  1980-2964  StripeTerminal          co.givealittle.kiosk                 I  class=StageEventLogger message="KernelProcessing newStatus IDLE event ContactCardUpdateEvent"
2023-09-25 10:44:24.349  1980-2964  StripeTerminal          co.givealittle.kiosk                 I  class=PaymentCollectionCoordinator message="onHandlePaymentCollectionEvent PresentPaymentMethodEvent(model=CollectionModel(transactionType=CHARGE, amount=Amount(value=1000, currency=GBP), baseAmount=Amount(value=1000, currency=GBP), callToAction=null, readerActiveInterfaces=[ICC, NFC], readerContactCardSlotState=EMPTY, readerEnableEditTip=false, enableCustomerCancellation=false))"
2023-09-25 10:44:24.351  1980-2964  StripeTerminal          co.givealittle.kiosk                 D  class=BbposAdapterPaymentCollectionListener message=onRequestPaymentMethodCollection
2023-09-25 10:44:24.352  1980-2964  StripeTerminal          co.givealittle.kiosk                 I  class=TerminalStatusManager message="requestReaderInput Insert / Tap."
2023-09-25 10:44:24.352  1980-2964  StripeTerminal          co.givealittle.kiosk                 I  class=ProxyReaderListener message="onRequestReaderInput(Insert / Tap)"
2023-09-25 10:44:24.352  1980-2964  StripePaymentSession    co.givealittle.kiosk                 D  onRequestReaderInput: Insert / Tap
2023-09-25 10:44:25.994  1980-1980  StripeTerminal          co.givealittle.kiosk                 I  class=BbposPaymentCollectionListener message="onRequestDisplayText SHOW_THANK_YOU"
```

**Comment by Kikimora** [comment]
Thanks for the report! Thinking of a workaround, how soon is too soon?

**Comment by mindy-stripe** [comment]
@superafroman apologies for the delayed response. From the logs you've provided, it looks like you are repeatedly calling `collectPaymentMethod`, but each of these calls should be followed up with a call to `confirmPaymentIntent` to confirm the payment intent, even when you are operating offline. 

Can you confirm the order of Terminal SDK calls that you are executing for these repeated calls to `collectPaymentMethod`?

**Comment by maggiewhite-stripe** [comment]
Hi @superafroman, thanks for reporting this issue! We shipped a fix for this issue for integrations that aren't using on-reader tipping in SDK version 3.1.0. If you are using [on-reader tipping](https://docs.stripe.com/terminal/features/collecting-tips/on-reader), that fix will be in the next upcoming release (3.4.0). I'll leave this ticket open until the fix is available for integrations that are using on-reader tipping in a published SDK release.

**Comment by billfinn-stripe** [comment]
Hi all -- we released 3.5.0 today. I'm going to close this issue, but feel free to reopen if you believe it hasn't been addressed.
